CONTRIBUTING.md

```diff
@@ -1,71 +1,63 @@
+# Contributing
+
+## Overview
+
+This documents explains the processes and practices recommended for contributing enhancements to
+this operator.
+
+- Generally, before developing enhancements to this charm, you should consider [opening an issue
+  ](https://github.com/canonical/mysql-k8s-operator/issues) explaining your use case.
+- If you would like to chat with us about your use-cases or proposed implementation, you can reach
+  us at [Canonical Mattermost public channel](https://chat.charmhub.io/charmhub/channels/charm-dev)
+  or [Discourse](https://discourse.charmhub.io/).
+- Familiarising yourself with the [Charmed Operator Framework](https://juju.is/docs/sdk) library
+  will help you a lot when working on new features or bug fixes.
+- All enhancements require review before being merged. Code review typically examines
+  - code quality
+  - test coverage
+  - user experience for Juju administrators of this charm.
+- Please help us out in ensuring easy to review branches by rebasing your pull request branch onto
+  the `main` branch. This also avoids merge commits and creates a linear Git commit history.
+
 ## Developing
 
-Create and activate a virtualenv with the development requirements:
+You can use the environments created by `tox` for development:
 
-```bash
-$ virtualenv -p python3 venv
-$ source venv/bin/activate
-$ pip install -r requirements-dev.txt
+```shell
+tox --notest -e unit
+source .tox/unit/bin/activate
 ```
 
-### Setup
-
-A typical setup using [snaps](https://snapcraft.io/), for deployments to a [microk8s](https://microk8s.io/) cluster can be done using the following commands
+### Testing
 
-```bash
-$ sudo snap install microk8s --classic
-$ microk8s.enable dns storage
-$ sudo snap install juju --classic
-$ juju bootstrap microk8s microk8s
-$ juju create-storage-pool operator-storage kubernetes storage-class=microk8s-hostpath
+```shell
+tox -e fmt           # update your code according to linting rules
+tox -e lint          # code style
+tox -e unit          # unit tests
+tox -e integration   # integration tests
+tox                  # runs 'lint' and 'unit' environments
 ```
 
-### Build
+## Build charm
 
-Install the charmcraft tool
+Build the charm in this git repository using:
 
-```bash
-$ sudo snap install charmcraft
+```shell
+charmcraft pack
 ```
 
-Build the charm in this git repository
-
-```bash
-$ charmcraft build
-```
-
-## Testing
-
-Unit tests are implemented using the Operator Framework test [harness](https://ops.readthedocs.io/en/latest/#module-ops.testing). These tests may executed by doing:
-
+### Deploy
 
 ```bash
-$ ./run_tests
+# Create a model
+juju add-model dev
+# Enable DEBUG logging
+juju model-config logging-config="<root>=INFO;unit=DEBUG"
+# Deploy the charm
+juju deploy ./mysql-k8s_ubuntu-20.04-amd64.charm \
+    --resource mysql-image=ubuntu/mysql --trust \
```

Comment:
I honestly should have paid more attention here - the `--trust` flag came from the operator template repository. Since we dont expect k8s API access, I removed the flag in e75502e92f3182ce1bb2789d4b2e67c7325e9ebb